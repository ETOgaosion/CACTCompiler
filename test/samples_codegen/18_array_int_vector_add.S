iobuff:
	.zero 80
.data
.globl	n
.type	n, @object
n:
	.word	0x6

.data
.globl	int_array_b
.type	int_array_b, @object
int_array_b:
	.word	0x7
	.word	0x8
	.word	0x9
	.word	0xa
	.word	0xb
	.word	0xc

.data
.globl	int_array_a
.type	int_array_a, @object
int_array_a:
	.word	0x1
	.word	0x2
	.word	0x3
	.word	0x4
	.word	0x5
	.word	0x6

ImmValue_2:
	.word	0x0
	.word	0x0
	.word	0x0
	.word	0x0
	.word	0x0
	.word	0x0

.text
.align 1
.globl	main
.type	main, @function
main:
	sd	ra, -8(sp)
	addi	sp, sp, -8
	call	int_array_vector_add
	addi	sp, sp, 8
	mv	a0, zero
	ld	ra, -8(sp)
	ret
