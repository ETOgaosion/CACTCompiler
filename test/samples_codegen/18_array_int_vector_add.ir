============ global var ================
n := symbol type: 2; data type: 2; initValue: n: 6

int_array_b := symbol type: 1; data type: 2; initValue: int_array_b: 7

int_array_a := symbol type: 1; data type: 2; initValue: int_array_a: 1

============ imm value ================
ImmValue_2: {0, 0, 0, 0, 0, 0}
============ functions ================
----------- funcDef -----------
int_array_vector_add() => 0:
----------- paramVars -----------
----------- localVars -----------
int_array_c := symbol type: 1; data type: 2; initValue: ImmValue_2: {0, 0, 0, 0, 0, 0}

----------- tempVars ------------
t_5:= symbol type: TEMP VAR; data type: 2
t_4:= symbol type: TEMP VAR; data type: 2
t_0:= symbol type: TEMP VAR; data type: 2
t_1:= symbol type: TEMP VAR; data type: 2
t_2:= symbol type: TEMP VAR; data type: 2
t_3:= symbol type: TEMP VAR; data type: 2
----------- Codes ------------
int_array_vector_add:
	int_array_c = ImmValue_2: {0, 0, 0, 0, 0, 0};
	t_0 = 0;
L_0_int_array_vector_add:
	t_1 = int_array_a[t_0];
	t_2 = int_array_b[t_0];
	t_3 = t_1 + t_2;
	int_array_c[t_0] = t_3;
	t_0 = t_0 + 1;
	t_4 = t_0 >= 6
	beqz t_4 L_0_int_array_vector_add
	t_5 = int_array_c[5];
	addparam t_5;
	call print_int, frame size: 56; 
	return ;

----------- funcDef -----------
main() => 2:
----------- paramVars -----------
----------- localVars -----------
----------- tempVars ------------
----------- Codes ------------
main:
	call int_array_vector_add, frame size: 8; 
	return 0;

