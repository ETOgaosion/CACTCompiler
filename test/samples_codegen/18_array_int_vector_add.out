bbl loader
couldn't open ELF program: test/samples_codegen/18_array_int_vector_add!
